tests/core/test_libsnappy_compat.py

```diff
@@ -20,3 +33,33 @@ def test_libsnappy_decompress_local_compressed(value):
     intermediate = compress(value)
     result = libsnappy_decompress(intermediate)
     assert value == result
+
+
+LIB_SNAPPY_ERRORS = (CompressedLengthError, UncompressError)
+PY_SNAPPY_ERRORS = (BaseSnappyError,)
+
+
+#
+# Error cases
+#
+@given(value=st.binary(min_size=1, max_size=MEGABYTE // 2))
+@settings(max_examples=400, deadline=None)  # takes a long time.
+def test_decompress_error_parity(value):
+    try:
+        py_result = decompress(value)
+    except PY_SNAPPY_ERRORS:
+        py_snappy_error = True
+    else:
+        py_snappy_error = False
+
+    try:
+        lib_result = libsnappy_decompress(value)
+    except LIB_SNAPPY_ERRORS:
+        lib_snappy_error = True
+    else:
+        lib_snappy_error = False
+
+    if py_snappy_error and lib_snappy_error:
+        pass
```

Comment:
We want to catch if one fails and the other doesn't, right? Something like:

```py
assert py_snappy_error is lib_snappy_error
```

Reply:
There isn't parity between the errors they two libraries raise, only that they either always both do or do not raise an exception.  In the case where they both raise (the case being checked here), this is the only assertion I think we can make.

Reply:
These are both just `bool`s, a direct comparison should work, no? As is, I don't see how the test catches if one library fails and the other doesn't.

Reply:
You're right, I'll update